pkg/generate/app/imagelookup_test.go

```diff
@@ -255,3 +256,72 @@ func fakeImageStream(name string, supports map[string]string) (*imageapi.ImageSt
 	}
 	return stream, images
 }
+
+func TestInputImageFromMatch(t *testing.T) {
+	tests := []struct {
+		name        string
+		match       *ComponentMatch
+		expectedTag string
+		expectedRef string
+	}{
+		{
+			name: "image stream",
+			match: &ComponentMatch{
+				ImageStream: &imageapi.ImageStream{
+					ObjectMeta: kapi.ObjectMeta{
+						Name:      "testimage",
+						Namespace: "myns",
+					},
+					Status: imageapi.ImageStreamStatus{
+						DockerImageRepository: "test/imagename",
+					},
+				},
+			},
+			expectedRef: "test/imagename:latest",
+		},
+		{
+			name: "image stream with tag",
+			match: &ComponentMatch{
+				ImageStream: &imageapi.ImageStream{
+					ObjectMeta: kapi.ObjectMeta{
+						Name:      "testimage",
+						Namespace: "myns",
+					},
+					Status: imageapi.ImageStreamStatus{
+						DockerImageRepository: "test/imagename",
+					},
+				},
+				ImageTag: "v2",
+			},
+			expectedRef: "test/imagename:v2",
+		},
+		{
+			name: "docker image",
+			match: &ComponentMatch{
+				Image: &imageapi.DockerImage{},
+				Value: "test/dockerimage",
+			},
+			expectedRef: "test/dockerimage",
```

Comment:
if you were looking for a docker image and we found it, we leave the ref alone. I can see that we'd want to add the :latest tag, but I believe that at some point you could have different default tags per repository, so not sure that there's a benefit in us adding latest